src/west/commands/project.py

```diff
@@ -277,27 +278,45 @@ def do_add_parser(self, parser_adder):
             - abspath: absolute and normalized path to the project
             - posixpath: like abspath, but in posix style, that is, with '/'
               as the separator character instead of '\\'
-            - revision: project's manifest revision
+            - revision: project's revision as it appears in the manifest
+            - sha: project's revision as a SHA
             - cloned: "(cloned)" if the project has been cloned, "(not cloned)"
               otherwise
             - clone_depth: project clone depth if specified, "None" otherwise
             '''.format(default_fmt)))
 
     def do_run(self, args, user_args):
+        def sha_thunk(project):
+            if project.revision:
+                return _sha(project, MANIFEST_REV)
+            else:
+                return '{:40}'.format('N/A')
+
+        def cloned_thunk(project):
+            return "(cloned)" if _cloned(project) else "(not cloned)"
+
+        def delay(func, project):
```

Comment:
Very clever solution to avoid retrieving the SHA or checking if a project is cloned, but for the sake of simplicity I ask myself if this is really needed. It's not like `west list` will ever be a time-critical operation? Also `_sha()` and `_cloned()` are not particularly slow?

Reply:
Git is so, so so slow on Windows :(